app/services/licences/fetch-licence-version-purpose-condition.service.js

```diff
@@ -10,45 +10,74 @@ const LicenceVersionPurposeCondtionModel = require('../../models/licence-version
 /**
  * Fetch the matching licence version purpose conditions and return data needed for the view licence page
  *
- * @param {string} id The UUID for the licence_version_purpose_id to fetch
+ * @param {object} ids The object contains the UUIDs for the licence_version_purpose_id and the purpose_id
  *
  * @returns {Promise<Object>} the data needed to populate the view licence page and some elements of the summary tab
  */
-async function go (id) {
-  if (!id) {
+async function go (ids) {
+  if (!ids) {
     return {
-      abstractionConditions: []
+      numberOfAbstractionConditions: 0,
+      uniqueAbstractionConditions: []
     }
   }
-  const LicenceVersionPurposeCondition = await _fetchLicenceVersionPurposeCondition(id)
-  const data = await _data(LicenceVersionPurposeCondition)
+
+  const licenceVersionIds = []
+
+  ids.forEach((item) => { licenceVersionIds.push(item.licenceVersionPurposeId) })
+
+  const LicenceVersionPurposeCondition = await _fetchLicenceVersionPurposeCondition(licenceVersionIds)
+  const data = await _data(LicenceVersionPurposeCondition, ids)
 
   return data
 }
 
-async function _data (licenceVersionPurposeConditions) {
+async function _data (licenceVersionPurposeConditions, licenceVersionPuroseAndPurposeUseIds) {
```

Comment:
Typo!

```suggestion
async function _data (licenceVersionPurposeConditions, licenceVersionPurposeAndPurposeUseIds) {
```

But I also think this function has grown too large and is doing too much. I would suggest moving the logic inside the loop into its own function that we call from here. That will also allow us to name the function something meaningful that might help explain what is going on (my little brain gets lost easily with what we're doing here!)